src/Debug/Profile.php

```diff
@@ -0,0 +1,239 @@
+    }
+
+    public static function getCurrent(): self
+    {
+        if (self::$current === null) {
+            $id = $_SERVER['HTTP_X_GLPI_AJAX_ID'] ?? bin2hex(random_bytes(8));
+            $parent_id = $_SERVER['HTTP_X_GLPI_AJAX_PARENT_ID'] ?? null;
+            self::$current = new self($id, $parent_id);
+
+            // If this is a sub-request, we can send the $_SERVER global data back in a header now to avoid having to save it later.
+            if ($parent_id !== null) {
+                try {
+                    $server_global = json_encode($_SERVER ?? [], JSON_THROW_ON_ERROR);
+                    $header_len = strlen($server_global);
+                    if ($header_len < self::DEBUG_INFO_HEADER_THRESHOLD) {
+                        header('X-GLPI-Debug-Server-Global: ' . $server_global);
+                    }
+                } catch (\Exception $e) {
+                    // Ignore
+                }
+            }
+
+            // Register a shutdown function to save the profile
+            register_shutdown_function(static function () {
+                // Stop all profiler timers (should just be the main php_request one unless something died)
+                Profiler::stopAll();
+                self::getCurrent()->save();
+            });
+        }
+        return self::$current;
+    }
+
+    public static function load(string $id, bool $delete = true): ?self
+    {
+        $profile_location = GLPI_TMP_DIR . '/debug/profiles/';
+        // only keep letters and numbers in the id to avoid path traversal
+        $id = preg_replace('/[^a-zA-Z0-9]/', '', $id);
+        $profile_name = $id . '.json.gz';
+
+        if (!file_exists($profile_location . $profile_name)) {
+            return null;
+        }
+
+        try {
+            $profile_data = json_decode(gzdecode(file_get_contents($profile_location . $profile_name)), true, 512, JSON_THROW_ON_ERROR);
+            $profile = new self($profile_data['id'], $profile_data['parent_id']);
+            $profile->is_readonly = true;
+            $profile->start_time = $profile_data['start_time'];
+            $profile->debug_info = $profile_data;
+
+            if ($delete) {
+                unlink($profile_location . $profile_name);
+            }
+            return $profile;
+        } catch (\Exception $e) {
+            return null;
+        }
+    }
+
+    public function getID(): string
+    {
+        return $this->id;
+    }
+
+    public function getParentID(): ?string
+    {
+        return $this->parent_id;
+    }
+
+    public function addData(string $widget, $data)
+    {
+        if (!array_key_exists($widget, $this->additional_info)) {
+            $this->additional_info[$widget] = [];
+        }
+        $this->additional_info[$widget][] = $data;
+    }
+
+    public function getDebugInfo()
+    {
+        if ($this->is_readonly) {
+            return $this->debug_info;
+        }
+        global $CFG_GLPI, $DEBUG_SQL, $SQL_TOTAL_REQUEST, $TIMER_DEBUG;
```

Comment:
We should probably keep `$DEBUG_SQL`, `$SQL_TOTAL_REQUEST` and `$TIMER_DEBUG` global variables in GLPI 10.0 to ensure we will not break any plugin, but IMHO, profiler should have methods like `startGlobalTimer()` and `addSqlQueryProfile()`, in order to be able to remove these global variables in GLPI 10.1.